[Documentation]: Adding example for Scalar Column Hydration (#9180)
doctrine/doctrine2: docs/en/reference/dql-doctrine-query-language.rst

Scalar Column Hydration
^^^^^^^^^^^^^^^^^^^^^^^

> If your query returns just **one column**, the result looks something like this:

Reviewer: ```suggestion
If your query returns **one column**, the result may look like this:
```

.. code-block:: php

    [
        ['id' => 5],
        ['id' => 12],
        ['id' => 23],
    ]

> In this case, you can use the scalar column hydration to reduce the result into a

Reviewer 2: ```suggestion
In this case, you can use the single scalar column hydration to reduce the result into a
```

> In this case, you can use scalar column hydration to "flatten" it into a

Reviewer: ```suggestion
In this case, you can use the scalar column hydration to reduce the result into a
```

one-dimensional array like this:

.. code-block:: php

    [5, 12, 23]

.. code-block:: php

>     <?php
>     use Doctrine\ORM\AbstractQuery;
> 
>     $query = $entityManager->createQuery('SELECT u.id FROM User u');
>     $ids = $query->getSingleColumnResult();
>     // Same as:
>     $ids = $query->getResult(AbstractQuery::HYDRATE_SCALAR_COLUMN);

Reviewer: ```suggestion
    <?php

    $query = $entityManager->createQuery('SELECT u.id FROM User u');
    $ids = $query->getSingleColumnResult();

or

.. code-block:: php

    <?php
    use Doctrine\ORM\AbstractQuery;

    $query = $entityManager->createQuery('SELECT u.id FROM User u');
    $ids = $query->getResult(AbstractQuery::HYDRATE_SCALAR_COLUMN);
```

Author: You mean because `use Doctrine\ORM\AbstractQuery;` is not needed in the first case? Is this really worth an entire code block?

Reviewer 2: Yes, it makes sense to have separate block, to be more clear when it is needed.
